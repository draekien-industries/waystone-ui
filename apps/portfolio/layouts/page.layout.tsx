/** @jsxImportSource theme-ui */
import React from 'react';
import Head from 'next/head';
import { TopBar, TopBarProps } from '../components';

export interface HeadProps {
  title: string;
  description: string;
}

export interface PageLayoutProps {
  headProps: HeadProps;
  topBarProps?: TopBarProps;
}

export const PageLayout = ({
  headProps,
  topBarProps,
  children,
}: React.PropsWithChildren<PageLayoutProps>) => (
  <>
    <Head>
      <title>{headProps.title}</title>
      <meta name="description" content={headProps.description} />
    </Head>
    <TopBar {...topBarProps} />
    <main sx={{ px: 'xl' }}>{children}</main>
  </>
);
