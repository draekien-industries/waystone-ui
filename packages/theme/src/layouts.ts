export const layouts = {
  radii: {
    sm: '0.25rem',
    md: '0.5rem',
    lg: '0.75rem',
    xl: '1rem',
    max: 999,
  },
  space: {
    xs: '0.25rem',
    sm: '0.5rem',
    md: '1rem',
    lg: '1.5rem',
    xl: '2rem',
  },
  zIndices: {
    lowest: 0,
    lower: 10,
    low: 20,
    middle: 30,
    high: 40,
    higher: 50,
    highest: 60,
    pagination: 1,
    navbar: 20,
    dropdown: 30,
    drawer: 40,
    modal: 50,
    toast: 60,
  },
} as const;

export default layouts;
