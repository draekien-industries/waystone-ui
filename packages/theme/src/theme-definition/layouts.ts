export const layouts = {
  radii: {
    small: '0.25rem',
    medium: '0.5rem',
    large: '0.75rem',
    'extra-large': '1rem',
    max: 999,
  },
  space: {
    'extra-small': '0.25rem',
    small: '0.5rem',
    medium: '1rem',
    large: '1.5rem',
    'extra-large': '2rem',
  },
  zIndices: {
    lowest: 0,
    lower: 10,
    low: 20,
    middle: 30,
    high: 40,
    higher: 50,
    highest: 60,
    pagination: 1,
    navbar: 20,
    dropdown: 30,
    drawer: 40,
    modal: 50,
    toast: 60,
  },
} as const;

export default layouts;
