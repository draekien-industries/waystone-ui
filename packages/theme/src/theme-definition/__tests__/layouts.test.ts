import { layouts } from '../layouts';

describe('layouts', () => {
  it('should match snapshot', () => {
    expect(layouts).toMatchSnapshot();
  });
});
