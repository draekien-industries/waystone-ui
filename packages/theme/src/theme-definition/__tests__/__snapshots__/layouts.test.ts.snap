// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts should match snapshot 1`] = `
{
  "radii": {
    "extra-large": "1rem",
    "large": "0.75rem",
    "max": 999,
    "medium": "0.5rem",
    "small": "0.25rem",
  },
  "space": {
    "extra-large": "2rem",
    "extra-small": "0.25rem",
    "large": "1.5rem",
    "medium": "1rem",
    "small": "0.5rem",
  },
  "zIndices": {
    "drawer": 40,
    "dropdown": 30,
    "high": 40,
    "higher": 50,
    "highest": 60,
    "low": 20,
    "lower": 10,
    "lowest": 0,
    "middle": 30,
    "modal": 50,
    "navbar": 20,
    "pagination": 1,
    "toast": 60,
  },
}
`;
